apps/ui/src/components/Ui/Address.vue

```diff
@@ -0,0 +1,43 @@
+<script setup lang="ts">
+import { shortenAddress } from '@/helpers/utils';
+
+withDefaults(
+  defineProps<{
+    address: string;
+    copyButton?: 'always' | 'hover';
+  }>(),
+  { copyButton: 'hover' }
+);
+
+const { copy, copied } = useClipboard();
+</script>
+
+<template>
+  <div class="flex items-center gap-2 group">
```

Comment:
This is slightly better
```suggestion
  <div class="flex items-center gap-1.5 group">
```